tests/problems.py

```diff
@@ -133,44 +132,127 @@ def nfe(self):
         return self._nfe
 
 
-class Ex4(BaseSDE):
-    noise_type = NOISE_TYPES.general
+class Neural1(BaseSDE):
+    noise_type = NOISE_TYPES.diagonal
+
+    def __init__(self, d, sde_type=SDE_TYPES.ito, **kwargs):
+        super(Neural1, self).__init__(sde_type=sde_type, noise_type=Neural1.noise_type)
+
+        self.f_net = nn.Sequential(
+            nn.Linear(d + 1, 8),
+            nn.Softplus(),
+            nn.Linear(8, d)
+        )
+        self.g_net = nn.Sequential(
+            nn.Linear(d + 1, 8),
+            nn.Softplus(),
+            nn.Linear(8, d),
+            nn.Sigmoid()
+        )
+
+    def f(self, t, y):
+        ty = torch.cat([t.expand(y.size(0), 1), y], dim=1)
+        return self.f_net(ty)
+
+    def g(self, t, y):
+        ty = torch.cat([t.expand(y.size(0), 1), y], dim=1)
+        return self.g_net(ty)
+
+    def h(self, t, y):
+        return torch.zeros_like(y)
+
+
+class Neural2(BaseSDE):
+    noise_type = NOISE_TYPES.scalar
+
+    def __init__(self, d, sde_type=SDE_TYPES.ito, **kwargs):
+        super(Neural2, self).__init__(sde_type=sde_type, noise_type=Neural2.noise_type)
+
+        self.f_net = nn.Sequential(
+            nn.Linear(d + 1, 8),
+            nn.Softplus(),
+            nn.Linear(8, d)
+        )
+        self.g_net = nn.Sequential(
+            nn.Linear(d + 1, 8),
+            nn.Softplus(),
+            nn.Linear(8, d),
+            nn.Sigmoid()
+        )
+
+    def f(self, t, y):
+        ty = torch.cat([t.expand(y.size(0), 1), y], dim=1)
+        return self.f_net(ty)
+
+    def g(self, t, y):
+        ty = torch.cat([t.expand(y.size(0), 1), y], dim=1)
+        return self.g_net(ty).unsqueeze(-1)
+
+    def h(self, t, y):
+        return torch.zeros_like(y)
+
+
+class Neural3(BaseSDE):
+    noise_type = NOISE_TYPES.additive
 
     def __init__(self, d, m, sde_type=SDE_TYPES.ito, **kwargs):
-        super(Ex4, self).__init__(sde_type=sde_type, noise_type=Ex4.noise_type)
-        self._nfe = 0
+        super(Neural3, self).__init__(sde_type=sde_type, noise_type=Neural3.noise_type)
```

Comment:
Git is choosing to interleave Ex4 and Neural3 here; Ex4 and Neural4 would have been a better match. (Just to help you understand why there seem to be lots of changes.)